src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSWebWorker.cs

```diff
@@ -14,9 +14,37 @@ namespace System.Runtime.InteropServices.JavaScript
     /// This is draft for possible public API of browser thread (web worker) dedicated to JS interop workloads.
     /// The method names are unique to make it easy to call them via reflection for now. All of them should be just `RunAsync` probably.
     /// </summary>
-    public static class WebWorker
+    public static class JSWebWorker
     {
-        public static Task<T> RunAsync<T>(Func<Task<T>> body, CancellationToken cancellationToken)
+        // temporary, for easy reflection
+        internal static Task RunAsyncVoid(Func<Task> body, CancellationToken cancellationToken) => RunAsync(body, cancellationToken);
+        internal static Task<T> RunAsyncGeneric<T>(Func<Task<T>> body, CancellationToken cancellationToken) => RunAsync(body, cancellationToken);
+
+        public static Task<T> RunAsync<T>(Func<Task<T>> body)
+        {
+            return RunAsync(body, CancellationToken.None);
+        }
+
+        public static Task RunAsync(Func<Task> body)
+        {
+            return RunAsync(body, CancellationToken.None);
+        }
+
+        public static async Task<T> RunAsync<T>(Func<Task<T>> body, CancellationToken cancellationToken)
+        {
+            // TODO remove main thread condition later
+            if (Thread.CurrentThread.ManagedThreadId == 1) await JavaScriptImports.ThreadAvailable().ConfigureAwait(false);
```

Comment:
```suggestion
            if (Thread.CurrentThread.ManagedThreadId == 1)
                await JavaScriptImports.ThreadAvailable().ConfigureAwait(false);
```

Reply:
thanks, I will fix that in next PR

Reply:
Fixed here https://github.com/dotnet/runtime/pull/95959